src/Storages/StorageMySQL.cpp

```diff
@@ -213,12 +213,13 @@ class StorageMySQLBlockOutputStream : public IBlockOutputStream
 
 BlockOutputStreamPtr StorageMySQL::write(const ASTPtr & /*query*/, const StorageMetadataPtr & metadata_snapshot, ContextPtr local_context)
 {
+    auto check_insert_privilege = getContext()->getSettingsRef().external_storage_check_insert_privilege && pool->getReplicasNumber() > 1 ? remote_table_name : "";
```

Comment:
```suggestion
    auto check_insert_privilege = getContext()->getSettingsRef().external_storage_check_insert_privilege ? remote_table_name : "";
```

It should work fine too (we only need to fix the error message)